### PR TITLE
Fix next tab activation in focus mode upon tab closing

### PR DESCRIPTION
**Pull request by Don-Vito**
## PR Checklist
* [x] Closes https://github.com/microsoft/terminal/issues/7916
* [x] CLA signed. 
* [ ] Tests added/passed
* [ ] Documentation updated. 
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. 

## Detailed Description of the Pull Request / Additional comments
Upon tab close the tabview is responsible to issue tab selection for the next active tab.
However this doesn't happen when tabview is hidden.
There was a special treatment for this scenario for full screen mode.
Added the same treatment to focus mode (as the tabview is not visible in this case as well).

## Validation Steps Performed
Manual tests

### CODE REVIEW

**Comment by Don-Vito** [comment]
@zadjii-msft - Happy Hanukkah!

**Comment by Don-Vito** [comment]
@zadjii-msft - not sure what static analyzer wants, I added a single word to the code 😄

**Comment by DHowett** [comment]
@don-vito (fixed in #8551)

**Review by zadjii-msft — APPROVED** [review]
WOW good catch. Thanks!

**Review by carlos-zamora — APPROVED** [review]

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -1136,13 +1136,16 @@ namespace winrt::TerminalApp::implementation
         {
             _lastTabClosedHandlers(*this, nullptr);
         }
-        else if (_isFullscreen || _rearranging)
+        else if (_isFullscreen || _rearranging || _isInFocusMode)
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TerminalPage.cpp:1139]: idle thought: _should this just be checking the visibility of TabView versus `Collapsed`? lol_

**Pull request merged**